.cursor/rules/claude-development-workflow.mdc

```diff
@@ -0,0 +1,198 @@
+---
+description: Claude AI assistant workflow patterns and best practices for Konflux UI development
+alwaysApply: true
+---
+
+# Claude Development Workflow
+
+> 📖 **Project Context**: This is Konflux UI, a React TypeScript application for Kubernetes CI/CD management.
+> 📖 **See Also**: [project-structure.mdc](mdc:.cursor/rules/project-structure.mdc) for architecture details.
+
+Guidelines for effective collaboration between developers and Claude AI assistant on Konflux UI.
+
+## Task Planning & Management
+
+### Always Use Todo Lists
+For any non-trivial task (3+ steps), Claude should create and maintain a todo list:
+
+```markdown
+✅ Research existing component patterns
+🔄 Implement new feature component  
+⏳ Add unit tests
+⏳ Update integration tests
+⏳ Run lint and typecheck
+```
+
+### Task Breakdown Patterns
+- **Feature Addition**: Research → Design → Implement → Test → Lint/Typecheck
+- **Bug Fix**: Reproduce → Identify → Fix → Test → Verify
+- **Refactoring**: Analyze → Plan → Execute → Test → Validate
+
+## Code Understanding Workflow
+
+### Before Making Changes
+1. **Search and understand** - Use search tools to understand existing patterns
+2. **Read related files** - Check imports, types, and similar components
+3. **Follow project conventions** - Match existing code style and architecture
+
+### Research Commands
+```bash
+# Find component patterns
+find src/components -name "*.tsx" | head -10
+
+# Search for similar functionality  
+grep -r "useK8sWatchResource" src/components/
+
+# Check test patterns
+find . -name "*.spec.tsx" -path "*/components/*"
+```
+
+## Development Best Practices
+
+### Component Development
+- **Check existing patterns** first in `src/components/` and `src/shared/components/`
+- **Use TypeScript types** from `src/types/` and `src/models/`
+- **Follow PatternFly conventions** for UI components
+- **Implement feature flags** for new features using `~/feature-flags/hooks`
+
+### Testing Requirements
+- **Create unit tests** using patterns from `testing.mdc`
+- **Use test utilities** from `~/unit-test-utils/`
+- **Mock K8s resources** with `createK8sWatchResourceMock`
+- **Test both success and error scenarios**
+
+### Code Quality Checks
+Always run these commands after code changes:
+```bash
+npm run lint
```

Comment:
we should use `yarn` commands instead of `npm` :)

Reply:
agree to change it to keep consistent with other md and good performance.